File: node.js/WorkingWithItems/batch-get.js
const AWS = require("aws-sdk");

AWS.config.update({ region: "us-west-2" });

const documentClient = new AWS.DynamoDB.DocumentClient();

// Define the partition keys and sort keys for the two items we want to get.
const params = {
  RequestItems: {
    RetailDatabase: {
      Keys: [
        {
          pk: "vikram.johnson@somewhere.com",
          sk: "metadata",
        },
        {
          pk: "jose.schneller@somewhere.com",
          sk: "metadata",
        },
      ],
      ConsistentRead: false, // For my user case, this data it is not changed often so why not get the reads at half price? Your use case might be different and need true.
    },
  },
  ReturnConsumedCapacity: "TOTAL",
};

documentClient.batchGet(params, (err, data) => {
  if (err) {
    console.error(JSON.stringify(err, null, 2));
  } else {
    console.log("BatchGetItem succeeded:", JSON.stringify(data, null, 2));
  }
});

File: golang/WorkingWithItems/batchGet/batchGet.go
package main

import (
    "github.com/aws/aws-sdk-go/aws"
	"github.com/aws/aws-sdk-go/aws/session"
	"github.com/aws/aws-sdk-go/service/dynamodb"
    "fmt"
)

var awsRegion = "us-west-2"

func getSession() (*session.Session) {
    sess := session.Must(session.NewSessionWithOptions(session.Options{
        SharedConfigState: session.SharedConfigEnable,
        // Provide SDK Config options, such as Region and Endpoint
        Config: aws.Config{
            Region: aws.String(awsRegion),
        },
    }))

    return sess
}

func getItems() error {
	dynamoDBClient := dynamodb.New(getSession())

	params := &dynamodb.BatchGetItemInput{
		RequestItems: map[string]*dynamodb.KeysAndAttributes{
            "Reply": {
                Keys: []map[string]*dynamodb.AttributeValue{
                    {
                        "pk": &dynamodb.AttributeValue{
                            S: aws.String("vikram.johnson@somewhere.com"),
                        },
                        "sk": &dynamodb.AttributeValue{
                            S: aws.String("metadata"),
                        },
                    },
                    {
                        "pk": &dynamodb.AttributeValue{
                            S: aws.String("jose.schneller@somewhere.com"),
                        },
                        "sk": &dynamodb.AttributeValue{
                            S: aws.String("metadata"),
                        },
                    },
                },
            },
		},
	}

	items, err := dynamoDBClient.BatchGetItem(params)

	if err != nil {
		fmt.Println("Error getting batch of items ...", err)
		return err
    }

    fmt.Println("Items ...")
    fmt.Println(items)

	return nil
}

func main() {
    fmt.Println("Getting Batch of Items ...")
    getItems()
    fmt.Println("Finished ...")
}
